lib/database.h

```diff
@@ -46,7 +46,7 @@ class QUOTIENT_API Database : public QObject
 
     // Returns a map UserId -> [DeviceId] that have not received key yet
     QHash<QString, QStringList> devicesWithoutKey(const QString& roomId, QHash<QString, QStringList>& devices, const QString &sessionId);
-    void setDevicesReceivedKey(const QString& roomId, const QHash<QString, QList<std::pair<QString, QString>>>& devices, const QString& sessionId, int index);
+    void setDevicesReceivedKey(const QString& roomId, const QVector<std::tuple<QString, QString, QString>>& devices, const QString& sessionId, int index);
```

Comment:
I asked to document the tuple if you don't make a structure - but ok, this class needs a separate documentation effort anyway.